gateway/coprocess.go

```diff
@@ -582,6 +597,38 @@ func (h *CustomMiddlewareResponseHook) HandleResponse(rw http.ResponseWriter, re
 	return nil
 }
 
+func syncHeadersAndMultiValueHeaders(headers map[string]string, multiValueHeaders []*coprocess.Header) []*coprocess.Header {
+	updatedMultiValueHeaders := []*coprocess.Header{}
+
+	for k, v := range headers {
+		found := false
+		for _, header := range multiValueHeaders {
```

Comment:
@sredxny why is this second loop/lookup necessary? We are appending to multiValueHeaders[k] the headers value.
Wouldn't something like
```
for k,v := headers{
 currentHeader := multiValueHeaders[k]
 currentHeader.Value = append(currentHeader.Value, []string{v}...) 
}
```
Do the same here?

Reply:
I see the point, however `multivalueHeaders` is not a map, but a pointer of `coprocess.Header` so we cannot do things as:
`multiValueHeaders[k]` but instead we must do things as `multivalueHeaders[any-position].key`